app/src/main/java/com/javahelp/frontend/activity/LoginActivity.java

```diff
@@ -58,6 +59,19 @@ protected void onCreate(Bundle savedInstanceState) {
         viewModel.getLoginResult().observe(this, this::updateOnLoginResult);
 
         binding.loginButton.setOnClickListener(this::loginClick);
+
+        binding.botCReg.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                startActivity(new Intent(LoginActivity.this, ClientRegistrationActivity.class));
+            }
+        });
+        binding.botPReg.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                startActivity(new Intent(LoginActivity.this, ProviderRegistrationActivity.class));
+            }
+        });
```

Comment:
These should be done with lambda expressions or method references. See line 61 for an example of a method reference, and I think there should be some lambda expressions further up in the file. It's just a lot more readable.

Reply:
That is actually good idea